great_expectations/self_check/util.py

```diff
@@ -1394,6 +1427,10 @@ def build_pandas_validator_with_data(
     context: Optional[DataContext] = None,
 ) -> Validator:
     batch = Batch(data=df, batch_definition=batch_definition)
+
+    if context is None:
+        context = build_in_memory_runtime_context()  # type: ignore[assignment]
```

Comment:
Why the `type: ignore` here?

Reply:
@cdkini Because of the conflict between `DataContext` declared previously and `BaseDataContext` returned here (it is safe to ignore for now).

Reply:
Instead of `DataContext`, why don't we annotate with an `AbstractDataContext`?